src/modules/taskLoader.js

```diff
@@ -1,14 +1,92 @@
-function TaskLoader(lists, tasks) {
-  tasks.forEach((task) => {
+/* eslint-disable no-use-before-define */
+import { stringifier } from './taskMethods.js';
+
+function TaskLoader(lists, tasks, taskEditor) {
+  // Remove tasks before re-rendering
+  while (lists.hasChildNodes()) {
+    lists.removeChild(lists.firstChild);
+  }
+  // Map tasks inside the task container
+  tasks.forEach((task, index) => {
     const list = document.createElement('li');
+    const form = document.createElement('form');
+    form.className = 'task-form';
     const checkBox = document.createElement('input');
     checkBox.type = 'checkbox';
-    const text = document.createElement('span');
-    text.innerHTML = task.string;
+    checkBox.checked = task.bool;
+    task.index = index;
+
+    // Add event listener for checkbox
+    checkBox.addEventListener('change', () => {
+      tasks[index].bool = !task.bool;
+      if (task.bool === true) {
+        text.style.textDecoration = 'line-through';
+        form.classList.remove('on');
+        dots.className = 'bi bi-three-dots-vertical';
+      } else {
+        text.style.textDecoration = 'none';
+        form.classList.remove('on');
+        dots.className = 'bi bi-three-dots-vertical';
+      }
+      stringifier(tasks);
+    });
+
+    const text = document.createElement('input');
+    text.className = 'task';
+    text.value = task.string;
+    if (task.bool === true) text.style.textDecoration = 'line-through';
+
+    // Add event listener for each added form
+    form.addEventListener('submit', (event) => {
+      event.preventDefault();
+      text.blur();
+      taskEditor(text, index, tasks);
```

Comment:
Kindly note that as per the instructions your indexes should match the items. meaning if i have one `1` tem on the array, its index is `1` as per the instructions
<img width="1676" alt="Screenshot 2022-11-02 at 22 10 48" src="https://user-images.githubusercontent.com/48546992/199584408-13cbfff9-e15c-417f-a31b-4b807376eb09.png">

As per the screenshot attached below, your indexes start from `zero` `0`
<img width="1664" alt="Screenshot 2022-11-02 at 22 29 58" src="https://user-images.githubusercontent.com/48546992/199585383-8f762db1-9a05-40a6-87b4-e417d869f6e0.png">